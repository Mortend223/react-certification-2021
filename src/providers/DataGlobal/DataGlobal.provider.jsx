import React, { createContext, useContext, useReducer } from 'react';
import PropTypes from 'prop-types';

// Providers
import { useAuth } from '../Auth/Auth.provider';
import { useUserContext } from '../DataUser/DataUser.provider';

const DataContext = createContext(null);

function DataReducer(state, action) {
  switch (action.type) {
    case 'updateSearch': {
      return { ...state, search: action.term };
    }
    case 'toggleModal': {
      return { ...state, isOpen: !state.isOpen };
    }
    case 'updateTheme': {
      return { ...state, isDark: !state.isDark };
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`);
    }
  }
}

function DataProvider({ children }) {
  const { authenticated } = useAuth();
  const { isDarkTheme, setTheme } = useUserContext();
  const [state, dispatch] = useReducer(DataReducer, {
    favoriteVideos: [],
    isDark: authenticated ? isDarkTheme : true,
    isOpen: false,
    search: 'Wizeline',
  });

  const onChangeInput = (value) => {
    dispatch({ type: 'updateSearch', term: value });
  };
  const toggleModal = () => {
    dispatch({ type: 'toggleModal' });
  };
  const toggleTheme = () => {
    if (authenticated) {
      setTheme(state.isDark);
    }
    dispatch({ type: 'updateTheme' });
  };

  return (
    <DataContext.Provider
      value={{
        search: state.search,
        isDark: state.isDark,
        isOpen: state.isOpen,
        onChangeInput,
        toggleModal,
        toggleTheme,
      }}
    >
      {children}
    </DataContext.Provider>
  );
}

function useData() {
  const context = useContext(DataContext);
  if (context === undefined) {
    throw new Error('useData must be used within a DataProvider');
  }
  return context;
}

DataProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { useData };
export default DataProvider;
