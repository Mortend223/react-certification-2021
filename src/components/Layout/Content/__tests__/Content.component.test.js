import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import ContentComponent from '../Content.component';

// Context
import { useData } from '../../../../providers/DataGlobal/DataGlobal.provider';

jest.mock('../../../../providers/DataGlobal/DataGlobal.provider');

describe('Content Component', () => {
  it('Should show ContentComponent', () => {
    const video = [
      {
        id: 'nmXMgqjQzls',
        title: 'Wizeline',
        description: 'Wizeline',
        thumbnailUrl: 'https://nmXMgqjQzls/mqdefault.jpg',
      },
    ];
    useData.mockReturnValue({
      search: 'Wizeline',
      isDark: true,
    });
    const container = render(
      <BrowserRouter>
        <ContentComponent videos={video} />
      </BrowserRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
