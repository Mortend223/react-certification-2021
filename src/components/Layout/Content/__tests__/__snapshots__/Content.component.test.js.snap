// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component Should show ContentComponent 1`] = `undefined`;
